File: Daily_Practice/198_House_Robber.py
class Solution:
    def rob(self, nums: list[int]) -> int:
        if len(nums) == 2:
            return 0
        else:
            li = []
            for i in range(0,2,1):
                count = 0
                for j in range(i,len(nums),2):
                    count += nums[j]
                li.append(count)
            return max(li)

nums = [1,2,3,1]
obj = Solution().rob(nums)
print(obj)

File: Daily_Practice/198_House_Robber.go
package main

import "fmt"

func rob(slice []int) int {
	temp := 0
	if len(slice) == 2 {
		return 0
	} else {
		for i := 0; i < 2; i++ {
			count := 0
			for j := i; j < len(slice); j += 2 {
				count += slice[j]
			}
			if count > temp {
				temp = count
			}
		}
	}
	return temp
}

func main() {
	var nums = []int{1, 2, 1}
	obj := rob(nums)
	fmt.Println(obj)
}
